src/tseda/vpages/ignn.py

```diff
@@ -122,25 +127,52 @@ def plot(self, haplotype=0):
         )
         return p
 
+    def check_inputs(self, inds):
+        try:
+            print(self.individual_id)
+            if (
+                not isinstance(self.individual_id, (int, float))
+                or self.individual_id < 0
+            ):
+                self.position_index_warning.visible = True
+                return None
+            else:
+                self.position_index_warning.visible = False
+                nodes = inds.loc[self.individual_id].nodes
+                return nodes
+        except KeyError:
+            self.position_index_warning.visible = True
+            return None
+
     @pn.depends("individual_id", "window_size")
     def __panel__(self, **params):
         inds = self.datastore.individuals_table.data.rx.value
         if self.individual_id is None:
-            return pn.pane.Markdown("")
-        nodes = inds.loc[self.individual_id].nodes
-        return pn.Column(
-            pn.pane.Markdown(f"## Individual id {self.individual_id}"),
-            self.warning_pane,
-            pn.pane.Markdown(f"### Haplotype 0 (sample id {nodes[0]})"),
-            self.plot(0),
-            pn.pane.Markdown(f"### Haplotype 1 (sample id {nodes[1]})"),
-            self.plot(1),
-        )
+            return pn.pane.Markdown(
+                "**Enter a valid sample id to see the GNN haplotype plot.**"
+            )
+        nodes = self.check_inputs(inds)
+        if nodes is not None:
+            return pn.Column(
+                pn.pane.HTML(
+                    "<h2 style='margin: 0;'>GNN Haplotype plot "
+                    f"- Individual id {self.individual_id}</h2>",
+                    sizing_mode="stretch_width",
+                ),
+                self.warning_pane,
+                pn.pane.Markdown(f"### Haplotype 0 (sample id {nodes[0]})"),
+                self.plot(0),
+                pn.pane.Markdown(f"### Haplotype 1 (sample id {nodes[1]})"),
+                self.plot(1),
+            )
+        else:
+            return pn.Column(pn.pane.Markdown(""))
 
     def sidebar(self):
         return pn.Card(
             self.param.individual_id,
             self.param.window_size,
+            self.position_index_warning,
```

Comment:
I think renaming `position_index` to `individual_id` would make sense in this module

Reply:
Fixing!